Assignment-3/app/src/App.js

```diff
@@ -0,0 +1,76 @@
+import React from 'react'
+import "./App.css";
+import Parent from './components/Parent';
+import ProductList from './components/ProductList';
+import MainPageRouter from './components/MainPageRouter';
+import RouterForProduct from './components/RouterForProduct';
+import ParentForTheme from './components/ParentForTheme';
+import ChildForTheme from './components/ChildForTheme';
+import  ProtectedRouter  from './components/Form/ProtectedRouter';
+import DashboardMainPage from './components/userDashboardComponents/DashboardMainPage';
+function App() {
+  return (
+    <div>
+      <p className="quest">
+        1.create two components: a parent component that acts as a provider and
+        a child component that consumes the context. Create a context to manage
+        a user's authentication status (logged in or out). Implement a login
+        button in the child component that, when clicked, updates the
+        authentication status in the context to "logged in." Display a message
+        in the child component based on the user's authentication status. If the
+        user is logged in, show "Welcome, [username]!" Otherwise, display
+        "Please log in."
+      </p>
+      <Parent />
+      <p className="quest">
+        2.Extend the previous application to demonstrate the use of nested
+        contexts. Create a new context to manage user preferences (e.g., theme
+        preference). Modify the parent component to provide both the
+        authentication and preferences contexts. Create a child component that
+        consumes both contexts. Allow the user to change their theme preference
+        (light/dark) using a button in the child component. Display the theme
+        preference in the UI and adjust the component's styling accordingly.
+      </p>
+      <ChildForTheme />
+
+      <p className="quest">
+        3.creat pages: Home and About. Implement routing using React Router DOM.
+        Create navigation links to switch between the Home and About pages.
+        Display appropriate content on each page. Add a "404 Not Found" page for
+        any invalid routes.
+      </p>
+      <MainPageRouter />
+      <p className="quest">
+        4.Enhance the previous application by adding protected routes that
+        require authentication. Create a login page with a form that accepts a
+        username and password. Implement authentication logic (e.g., hardcode a
+        username and password for now). Secure the About page so that it can
+        only be accessed by authenticated users. Redirect unauthenticated users
+        to the login page. Display a message on the Home page welcoming the
+        authenticated user.
+      </p>
+      {/* <ProtectedRouter /> */}
```

Comment:
Task 4 is commented, why?

Reply:
it is protected router when we will use then automatically we will move to login page of question-4
so other router will not work properly